src/main/java/com/google/sps/data/HiddenGem.java

```diff
@@ -14,33 +14,63 @@
 public final class HiddenGem {
 
-  private final int id;
+  private final String placeId;
   public final String name;
-  public final String business_type;
+  public final String[] types;
   public final String address;
-  public final int price_level;
-  public final double rating;
-  public final String photo;
+  public final double lat;
+  public final double lng;
+  public final String price_level;
+  public final float rating;
+  public final int userRatingsTotal;
+  public final URL website;
+  public final OpeningHours openingHours;
+  public final String photoReference;
+  public final String htmlAttributions;
+  public final boolean permanentlyClosed;
+  public final String businessStatus;
 
   /** Create a new Hidden Gem. 
-   *  @param id             the id the of the hidden gem.
-   *  @param name           the name of the hidden gem.
-   *  @param business_type  the business type of the hidden gem, either restaurant or cafe.
-   *  @param address        the address of the hidden gem.
-   *  @param price_level    the price range of the hidden gem. From 0 (free) to 4 (very expensive).
-   *  @param rating         the google star rating of the hidden gem. From 1 (poor rating) to 5 (excellent rating).
-   *  @param photo          the photo url of the hidden gem.
+   *  @param id                the id the of the hidden gem.
+   *  @param name              the name of the hidden gem.
+   *  @param types             the business types of the hidden gem, e.g. restaurant or cafe.
+   *  @param address           the address of the hidden gem.
+   *  @param lat               the latitude of the hidden gem's location.
+   *  @param lng               the longitude of the hidden gem's location.
+   *  @param price_level       the price range of the hidden gem. From 0 (free) to 4 (very expensive).
+   *  @param rating            the google star rating of the hidden gem. From 1 (poor rating) to 5 (excellent rating).
+   *  @param userRatingsTotal  the total number of ratings provided by user's.
+   *  @param website           the website of the hidden gem.
+   *  @param openingHours      the hours when the hidden gem is open now.
+   *  @param photoReference    the photo url of the hidden gem.
+   *  @param htmlAttributions  the attributions about the hidden gem which must be displayed to the user.
+   *  @param permanentlyClosed the state is true if the hidden gem has permanently closed.
+   *  @param businessStatus    the current status of the hidden gem   
   */
-  public HiddenGem(int id, String name, String business_type, String address, int price_level, double rating, String photo) {
-    this.id = id;
-    this.name = name;
-    this.business_type = business_type;
-    this.address = address;
-    this.price_level = price_level;
-    this.rating = rating;
-    this.photo = photo;
+  public HiddenGem(String placeId, String name, String[] types, String address, double lat, 
```

Comment:
it might also be useful to have a second constructor that takes in the nearby search result and constructs a hidden gem?

Reply:
I currently construct the hidden gem object once I have both the nearby search (12 values) and place detail results (three values). 

I'm not sure I have understood correctly. With a second constructor with just the 12 parameters from the nearby Search api call - build an initial list of hidden gems with just the nearby search result. Then when it would come to adding the three additional details would i look through the list of hidden gems and use their values to construct another hidden gem with the additional values to replace it?

Reply:
Ahh sorry, didn't realise this was a mix of both data when it was first constructed. You could still create a contructor that takes both the nearby search results and the place detail results and populates the parameters accordingly. That way your class is taking responsibility for doing the conversion and you don't need the calling code to figure out how to do it. This is quite a long list of parameters so sometimes it's nice to have a simpler way of creating the object.